### ISSUE TITLE
Update for Decorative Pack 3?

### ISSUE THREAD

**Issue by HappypsychoX** [Lamer87/Space_Engineers_DLC_unlocker#36]
Will you be updating this for the DLC pack released today?

**Comment by Lamer87** [comment]
I'm downloading :) check the other unlockers linked in readme till I update.

**Comment by Lamer87** [comment]
It's ready ;)

**Comment by Superdie02** [comment]
it doesnt work for me

**Comment by Lamer87** [comment]
try redownloading, use the folder inside "dlc unlocker" not "original backups"

**Comment by Lamer87** [comment]
Everything is fine, there are no errors, if you downloaded it before you have to re-download it because it was uploading

**Comment by Superdie02** [comment]
can you check from the one you uploaded?

**Comment by HappypsychoX** [comment]
Can confirm, not working. Tried new game in creative..

**Comment by Superdie02** [comment]
could you try re-uploading it?
im willing to screen share on discord to prove it

discord: superjafar33

**Comment by Superdie02** [comment]
here is a photo of how it looks
![20230901002754_1](https://github.com/Lamer87/Space_Engineers_DLC_unlocker/assets/84743200/ba36ef4b-b3e5-4f29-860d-727da839d449)

**Comment by Lamer87** [comment]
Yes guys, I just tried ingame, the usual unlocking method doesn't seem to work anymore.
I'm investigating the situation.

**Comment by Superdie02** [comment]
when you figure out what is causing the issue, could you notify me since im trying to do stuff like this on my own?

**Comment by HappypsychoX** [comment]
In any case, I appreciate the quick response to the issue. I've been looking through the game files that were updated. I haven't seen anything that seems relevant to fixing this. I wonder if they buried the dlc check in a dll to make it harder to unlock.

**Comment by DOGGO-GOODBOY** [comment]
Does anyone know anyone else out there also working on this? I'm interested in what is it they've changed, and I also like learning about this kind of stuff in general

**Comment by Lamer87** [comment]
> I wonder if they buried the dlc check in a dll to make it harder to unlock

I believe this is exactly the case, as with the "emotes" and the "armor skins" now the DLCs are also blocked behind steam check.

Anyway guys, I found the solution, but it requires the use of "CreamAPI"...
In practice it's almost like a "crack" of the game, but I just tried it and everything works perfectly (I have the original)
it does not require .exe files, just the addition of 2 files, now I try to create a mini guide and upload the necessary files with related backups.

![Ah shit, here we go again](https://github.com/Lamer87/Space_Engineers_DLC_unlocker/assets/65097384/38b94217-cdc5-4844-a633-d897a0be917c)

Just tested on my original game, this is an old method that can "break" with every SINGLE game update (even small patches) but it's the only way left that works.

**Comment by Superdie02** [comment]
> 

where is this cream API
?

**Comment by DOGGO-GOODBOY** [comment]
Damn... I can't believe they "fixed" it once and for all. Thanks for your work, Lamer. 

I think I will resort to CreamAPI, I've known about it but some time but have never used it, so a rough guide would be welcome by all of us I'm sure.

**Comment by Superdie02** [comment]
do you know how to use creamAPI on space engineers?

**Comment by Lamer87** [comment]
just wait guys, I need to create a mini guide on how to (that works for newbies).

in short:
1 - C:\Program Files (x86)\Steam\steamapps\common\SpaceEngineers\Bin64\
2 - rename "steam_api64.dll" to "steam_api64_o.dll" ("o", not zero)
3 - then place 2 files in the game folder
4 - play.

to go back:
1- delete the 2 added files and rename the file back "steam_api64_o.dll" to "steam_api64.dll".

**Comment by Superdie02** [comment]
do i put the smoke API.dll's into the bin64?

**Comment by Lamer87** [comment]
[Space Engineers - CreapAPI for 1.203 v0.1](https://mega.nz/file/bB5kDYyR#c26LrGLC9ECuTsPd2LuJsOrw9PVBIa0rmnYP8C7B0eA)

Throw these files in the folder and play, there is already a separate backup of the "steam_api64.dll" file in case you need it.

in this archive:
- cream_api.ini : the config file
- steam_api64.dll : the creamapi file
- steam_api64_o.dll : the original "steam_api64.dll" but renamed (creamapi NEED it)
- backup\steam_api64.dll : just the original file NOT renamed, so you can just delete the other 3 and restore this one.

**Comment by HappypsychoX** [comment]
Confirmed working. You are awesome!

**Comment by Superdie02** [comment]
thank you mr lamar from the video game "grand theft auto online"

**Comment by Lamer87** [comment]
It's lamEr, because i'm a lamer, not lamar xD

**Comment by Felty871** [comment]
i have checked the x86 files and i cant find the referenced steam_api64.dll

**Comment by Lamer87** [comment]
SORRY, you have to put files in
C:\Program Files (x86)\Steam\steamapps\common\SpaceEngineers\Bin64\

**Comment by DOGGO-GOODBOY** [comment]
Looks like its working for me also, DLC blocks are back to being unlocked. You are doing Clang's work, you lovely Italian man.

Out of curiosity, is there a CreamAPI method that unlocks armor skins? I imagine suit skins are not possible since they're handled as actual tradeable Steam items, but does that go the same for armor paints?

**Comment by Felty871** [comment]
ohhhhhhhh

**Comment by Lamer87** [comment]
> is there a CreamAPI method that unlocks armor skins?

Omg I was looking for just this, so emotes and armor skins are also blocked with creamapi, and sorry but no, there is no method to do this.

**Comment by Felty871** [comment]
thank you so much!

**Comment by DOGGO-GOODBOY** [comment]
> and sorry but no, there is no method to do this.

Yeah, that's the conclusion I came to as well, just wanted to double-check. Thanks for your help again!

**Comment by Felty871** [comment]
hey im trying to install cream api and its setting my antivirus off like crazy
halp?

**Comment by Lamer87** [comment]
> hey im trying to install cream api and its setting my antivirus off like crazy

I have Avast, didn't tell me anything, just close your or make an exception into it.

**Comment by Felty871** [comment]
its windows defender

**Comment by Felty871** [comment]
windows is saying its a trogan horse

**Comment by Lamer87** [comment]
Just make an exception, this is like a crack of a game, and I think big mamasoft does not like it ;)

**Comment by DOGGO-GOODBOY** [comment]
@Felty871 Windows Defender thinks strange .dll files will implode your PC and kill your family. Ignore it. Let the files out of quarantine and make an exception. I'm playing right now using the files provided by Lamer

**Comment by Felty871** [comment]
alr tbh windows defender is wonky

**Comment by Lamer87** [comment]
Software houses don't like it when you use their games for free, and that's why the "trojan" gets detected ;)

**Comment by Felty871** [comment]
ahhh thanks :)
